### ISSUE TITLE
WooCommerce 5.6.0: default pages installed and activated after update

### ISSUE THREAD

**Issue by kosiew** [woothemes/woocommerce#30532]
### Steps to Reproduce

1.  Start with Woo below 5.6.0
2.  Delete one or more of the default store pages
3.  Update Woo to 5.6.0
4.  Allow some time for DB updates etc.

### What I expected to happen

I only expected the new Refund and Returns Policy page to be installed as a draft (it was) but none of the other pages to be impacted.

I would expect that we may create those other pages as drafts, but not publish them since not all store owners want this.

### What actually happened

The default pages are reinstalled and published

### Context

p9F6qB-7lE-p2

### Reproducibility

Consistent

### Other information

Worth noting, this will likely impact all Woo sites since I tested this on WPcom and also on a local test site with the same results.

**Comment by tammullen** [comment]
Thanks @kosiew for reporting this.

I haven't been able to reproduce this yet.

I setup a JN site 

![Screenshot 2021-08-19 at 10 11 46](https://user-images.githubusercontent.com/24649833/130042518-0357d9ed-5713-4e47-b341-187e979dbf1a.png)

Installed and activated 5.5.2 (I also tried with 5.4.2)

![Screenshot 2021-08-19 at 10 11 53](https://user-images.githubusercontent.com/24649833/130042571-bff4fd07-718a-4711-9b4a-9d5dadfb54dd.png)

I created two new pages.

![Screenshot 2021-08-19 at 10 12 01](https://user-images.githubusercontent.com/24649833/130042661-a18f6483-b85b-4971-8764-8a2878cb9e0d.png)

I then set these new pages to be the Shop page and My Account page

![Screenshot 2021-08-19 at 09 47 25](https://user-images.githubusercontent.com/24649833/130042792-1b27deb6-a690-4b17-afee-97cfa7a0da0d.png)
![Screenshot 2021-08-19 at 09 47 41](https://user-images.githubusercontent.com/24649833/130042806-94f1c8f5-360e-4a92-8b99-71d914ba813a.png)
![Screenshot 2021-08-19 at 10 12 11](https://user-images.githubusercontent.com/24649833/130042843-7fb8b0f1-b847-414b-b778-a1516ee680f8.png)

I then trashed and deleted the original `Shop` and `My Account` pages (I repeated this test with only trashing and had the same result)

![Screenshot 2021-08-19 at 10 12 20](https://user-images.githubusercontent.com/24649833/130042958-2499e2b8-c6b7-4da6-80eb-456388e4ecb8.png)

I then upgraded to 5.6.0, no new pages created before updating DB

![Screenshot 2021-08-19 at 10 12 20](https://user-images.githubusercontent.com/24649833/130043037-6002763a-a8e7-460d-83c9-116796376cdc.png)

I then upgraded the DB and only the new refund page was created in Draft

![Screenshot 2021-08-19 at 09 21 28](https://user-images.githubusercontent.com/24649833/130043104-9097c09f-eadf-4f43-8b20-90b5fbf1eb86.png)
![Screenshot 2021-08-19 at 09 23 00](https://user-images.githubusercontent.com/24649833/130043122-61aea577-a642-410d-86b2-2a8a3561a754.png)
![Screenshot 2021-08-19 at 10 12 27](https://user-images.githubusercontent.com/24649833/130043136-cab6ad1d-59a7-4d07-a029-efedfa6fb421.png)

I will try again now on some other types of site.

Can you share the System Status Report of any sites you are seeing this happen on?

**Comment by kosiew** [comment]
@tammullen ,
Thanks for the quick action.

Here is the SSR:

`
### WordPress Environment ###

WordPress address (URL): https://boekweitthee.nl
Site address (URL): https://boekweitthee.nl
WC Version: 5.6.0
REST API Version: ✔ 5.6.0
WC Blocks Version: ✔ 5.5.1
Action Scheduler Version: ✔ 3.2.1
WC Admin Version: ✔ 2.5.1
Log Directory Writable: ✔
WP Version: 5.8
WP Multisite: –
WP Memory Limit: 256MB
WP Debug Mode: ✔
WP Cron: ✔
Language: nl_NL
External object cache: ✔

### Server Environment ###

Server Info: nginx
PHP Version: 7.4.21
PHP Post Max Size: 2GB
PHP Time Limit: 1200
PHP Max Input Vars: 6144
cURL Version: 7.77.0
OpenSSL/1.1.0l

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.3.31-MariaDB log
Max Upload Size: 2GB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 5.6.0
WC Database Prefix: wp_
Total database size: 12.48MB
Database data size: 9.69MB
Database index size: 2.79MB
wp_woocommerce_sessions: Data: 0.22MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.11MB + Index: 0.13MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.08MB + Index: 0.06MB + Engine InnoDB
wp_ce4wp_abandoned_checkout: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_ce4wp_contacts: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_gla_budget_recommendations: Data: 0.22MB + Index: 0.14MB + Engine InnoDB
wp_gla_merchant_issues: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_gla_shipping_rates: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_gla_shipping_times: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailchimp_carts: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailchimp_jobs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailpoet_custom_fields: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_dynamic_segment_filters: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_feature_flags: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_forms: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailpoet_log: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailpoet_mapping_to_external_entities: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletters: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_newsletter_links: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_mailpoet_newsletter_option: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletter_option_fields: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletter_posts: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletter_segment: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_newsletter_templates: Data: 2.52MB + Index: 0.00MB + Engine InnoDB
wp_mailpoet_scheduled_tasks: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_scheduled_task_subscribers: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_segments: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_sending_queues: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_settings: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_statistics_clicks: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_mailpoet_statistics_forms: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_statistics_newsletters: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_statistics_opens: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
wp_mailpoet_statistics_unsubscribes: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_mailpoet_statistics_woocommerce_purchases: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_mailpoet_stats_notifications: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_subscribers: Data: 0.02MB + Index: 0.13MB + Engine InnoDB
wp_mailpoet_subscriber_custom_field: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_subscriber_ips: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailpoet_subscriber_segment: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_mailpoet_user_flags: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 1.22MB + Index: 0.06MB + Engine InnoDB
wp_postmeta: Data: 1.52MB + Index: 0.20MB + Engine InnoDB
wp_posts: Data: 2.31MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 118
customize_changeset: 25
custom_css: 1
feedback: 3
jetpack_migration: 2
jp_img_sitemap: 3
jp_pay_order: 2
jp_pay_product: 4
jp_sitemap: 3
jp_sitemap_master: 3
mailpoet_page: 1
nav_menu_item: 10
page: 12
post: 4
product: 23
product_variation: 16
revision: 182
shop_coupon: 1
wp_template_part: 2

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (16) ###

Akismet Anti-Spam: door Automattic – 4.1.10
AMP: door AMP Project Contributors – 2.1.3
CoBlocks: door GoDaddy – 2.16.0
Creative Mail by Constant Contact: door Constant Contact – 1.4.3
Crowdsignal Forms: door Automattic – 1.5.4
Facebook for WooCommerce: door Facebook – 2.6.1
Editing-toolkit voor WordPress.com: door Automattic – 3.12771
Google Listings and Ads: door WooCommerce – 1.4.1
Gutenberg: door Gutenberg Team – 11.3.0
Jetpack: door Automattic – 10.0
Layout Grid: door Automattic – 1.7
Mailchimp voor WooCommerce: door Mailchimp – 2.5.2
MailPoet 3 (New): door MailPoet – 3.67.0
Page Optimize: door Automattic – 0.5.1
WooCommerce Shipping & Tax: door WooCommerce – 1.25.18
WooCommerce: door Automattic – 5.6.0

### Inactive Plugins (1) ###

Classic Editor: door WordPress Contributors – 1.6.2

### Dropin Plugins (2) ###

advanced-cache.php: advanced-cache.php
object-cache.php: Memcached

### Must Use Plugins (1) ###

WP.com Site Helper: door  –

### Settings ###

API Enabled: –
Force SSL: –
Currency: EUR (€)
Currency Position: left_space
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Winkelbasis: #566 - /winkel/
Winkelmand: #567 - /winkelwagen/
Afrekenen: #568 - /afrekenen/
Mijn account: #569 - /mijn-account/
Algemene voorwaarden: ❌ Pagina niet ingesteld

### Theme ###

Name: Hever
Version: 1.5.11
Author URL: https://automattic.com/
Child Theme: ✔
Parent Theme Name: Varia
Parent Theme Version: 1.6.10
Parent Theme Author URL: https://automattic.com/
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

rounded: 275
Oldest: 2021-07-19 14:01:29 +0200
Newest: 2021-08-19 05:00:21 +0200

Pending: 3
Oldest: 2021-08-19 11:27:52 +0200
Newest: 2021-08-19 11:28:09 +0200


### Status report information ###

Generated at: 2021-08-19 11:28:09 +02:00
`

**Comment by tammullen** [comment]
Thanks @kosiew 

I have found that if I exclude the step of setting the new page I can reproduce this.

- Delete `My Account` page in 5.5.2

![Screenshot 2021-08-19 at 10 40 39](https://user-images.githubusercontent.com/24649833/130046619-7662f112-aee7-43b0-855b-7a1f6cf2c2ae.png)

- No page set 
![Screenshot 2021-08-19 at 10 31 03](https://user-images.githubusercontent.com/24649833/130046284-6d52f8b3-ac90-4fda-af2f-2c3c7d13b41b.png)

- Upgrade to 5.6.0 and update DB. `My Account` is recreated and published and set to be the My Account page.
![Screenshot 2021-08-19 at 10 39 57](https://user-images.githubusercontent.com/24649833/130046634-6ba42473-fdab-4fb9-8ec7-4b331673b346.png)

![Screenshot 2021-08-19 at 10 31 13](https://user-images.githubusercontent.com/24649833/130046303-fd0d98b2-f99b-426b-8279-1798e40e6fbf.png)



FWIW I also tried same steps in 4.8.1 and then upgraded to 5.0.1 and updated the DB (As this was the previous time we had a DB update I believe) and the page wasn't recreated.

I also checked these steps upgrading from 5.4.2 to 5.5.2 and the page wasn't recreated.

Priority of this issue will be assigned shortly.
